### PR TITLE
Feature: Maven now builds a distribution zip file, which is also persisted by circleci

### PR DESCRIPTION
**Pull request by Zottel92**
Now Maven creates a zip file again that is runable via a included script.

Issue: https://github.com/Zottel92/opennms-reloadable-daemon-poc/issues/20

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -121,8 +132,92 @@
...
+                <version>3.1.0</version>
+                <executions>
+                    <execution>
+                        <id>copy-resources</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${basedir}/target/assemblyFolder</outputDirectory>
```
Comment (mvrueden) [inline pom.xml]: Please use `assembly-folder` or something similar, but no camel case, as this is annoying, when typing folders/file names.

File: src/assembly/assembly.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<assembly
+        xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2 http://maven.apache.org/xsd/assembly-1.1.2.xsd">
+    <formats>
+        <format>zip</format>
+    </formats>
+    <id>package</id>
+    <includeBaseDirectory>false</includeBaseDirectory>
```
Comment (mvrueden) [inline src/assembly/assembly.xml]: The run.sh should be executable, so the file modifiers should be modified here.

File: src/assembly/assembly.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <files>
+        <file><outputDirectory>.</outputDirectory><source>target/assemblyFolder/jetty-runner.jar</source></file>
+        <file><outputDirectory>.</outputDirectory><source>target/assemblyFolder/run.sh</source></file>
+        <file><outputDirectory>.</outputDirectory><source>target/assemblyFolder/spring-mvc-jetty-example.war</source></file>
```
Comment (mvrueden) [inline src/assembly/assembly.xml]: This file does not exist.

File: pom.xml [hunk 4]
```diff
@@ -121,8 +132,92 @@
...
+                            <artifactId>${project.artifactId}</artifactId>
+                            <version>${project.version}</version>
+                            <type>war</type>
+                            <overWrite>false</overWrite>
+                            <outputDirectory>${project.build.directory}/assemblyFolder</outputDirectory>
+                            <destFileName>${project.name}.war</destFileName>
+                        </artifactItem>
+                    </artifactItems>
+                    <outputDirectory>${project.build.directory}/wars</outputDirectory>
+                    <overWriteReleases>false</overWriteReleases>
```
Comment (mvrueden) [inline pom.xml]: Should also be true

File: pom.xml [hunk 5]
```diff
@@ -121,8 +132,92 @@
...
+                            <overWrite>false</overWrite>
+                            <outputDirectory>${project.build.directory}/assemblyFolder</outputDirectory>
+                            <destFileName>jetty-runner.jar</destFileName>
+                        </artifactItem>
+                        <artifactItem>
+                            <groupId>${project.groupId}</groupId>
+                            <artifactId>${project.artifactId}</artifactId>
+                            <version>${project.version}</version>
+                            <type>war</type>
+                            <overWrite>false</overWrite>
```
Comment (mvrueden) [inline pom.xml]: Should probably true as well.

File: pom.xml [hunk 6]
```diff
@@ -104,6 +107,14 @@
             <scope>test</scope>
         </dependency>
 
+        <!-- jettey -->
+        <dependency>
```
Comment (mvrueden) [inline pom.xml]: A short description why this is necessary would be nice as well.

File: .circleci/config.yml [hunk 7]
```diff
@@ -47,3 +47,12 @@ jobs:
       - store_test_results:
           path: target/surefire-reports
 
+      # collect artifacts
+      - run:
+          name: Creating Artifacts Folder
```
Comment (mvrueden) [inline .circleci/config.yml]: I'd prefer something like `Collecting Artifacts`.

**Comment by Zottel92** [comment]
Updated pls review

**Pull request merged**